test/org/zaproxy/zap/authentication/UsernamePasswordAuthenticationCredentialsTest.java

```diff
@@ -0,0 +1,115 @@
+    }
+
+    @Test
+    public void shouldNotUsernameBeEmptyOrNull() throws Exception {
+        // Given
+        String username = usernamePasswordAuthenticationCredentials.getUsername();
+
+        // When/Then
+        assertThat(username, not(isEmptyOrNullString()));
+    }
+
+    @Test
+    public void shouldNotPasswordBeEmptyOrNull() throws Exception {
+        // Given
+        String password = usernamePasswordAuthenticationCredentials.getPassword();
+
+        // When/Then
+        assertThat(password, not(isEmptyOrNullString()));
+    }
+
+    @Test
+    public void shouldReturnTureIfIsConfigured() throws Exception {
+        // Given
+        boolean isConfigured = usernamePasswordAuthenticationCredentials.isConfigured();
+
+        // When/Then
+        assertThat(isConfigured, is(true));
+    }
+
+    @Test
+    public void shouldEncodeReturnNullCredentialsIfUsernameIsNull() throws Exception {
+        // Given
+        username = null;
+        password = "something";
+        String parentStringSeparator = "|";
+        usernamePasswordAuthenticationCredentials = new UsernamePasswordAuthenticationCredentials(username, password);
+
+        // When
+        String encodedResult = usernamePasswordAuthenticationCredentials.encode(parentStringSeparator);
+
+        // Then
+        assertThat(encodedResult, is("AA=="));
+    }
+
+    @Test(expected = AssertionError.class)
+    public void shouldParentStringSeparatorNotBeFieldSeparatorWhenEncode() throws Exception {
+        // Given
+        String parentStringSeparator = "~";
+        usernamePasswordAuthenticationCredentials = new UsernamePasswordAuthenticationCredentials(username, password);
+
+        // When
+        usernamePasswordAuthenticationCredentials.encode(parentStringSeparator);
+
+        // Then = AssertionError
+    }
+
+    @Test
+    public void shouldEncodeUsernameAndPasswordWithTheRightFieldSeparator() throws Exception {
+        // Given
+        String parentStringSeparator = "|";
+
+        // When
+        String encodedUsernamePassword = usernamePasswordAuthenticationCredentials.encode(parentStringSeparator);
+
+        // Then
+        assertThat(encodedUsernamePassword, not(isEmptyOrNullString()));
+        assertThat(encodedUsernamePassword, allOf(containsString("dXNlcm5hbWU="),
+                                                    containsString("cGFzc3dvcmQ="),
+                                                    containsString("~")));
+    }
+
+    @Test
+    public void shouldApiResponseRepresentationReturnApiResponseWithValidNameAndJsonFormat() throws Exception {
+        // Given
+        ApiResponse apiResponse = usernamePasswordAuthenticationCredentials.getApiResponseRepresentation();
+
+        // When/Then
+        assertThat(apiResponse.getName(), not(isEmptyOrNullString()));
+        assertThat(apiResponse.toJSON(), notNullValue());
+    }
+
+    @Test
+    public void shouldApiResponseRepresentationCallApiResponseSet() throws Exception {
```

Comment:
Does it worth verify which `ApiResponse` implementation is being used? (That can be changed while still returning the expected data/format.)